Zend/zend_operators.c

```diff
@@ -2057,16 +2057,17 @@ has_op2_string:;
 		}
 
 		if (result == op1) {
-			/* special case, perform operations on result */
-			result_str = zend_string_extend(op1_string, result_len, 0);
-			/* Free result after zend_string_extend(), as it may throw an out-of-memory error. If we
-			 * free it before we would leave the released variable on the stack with shutdown trying
-			 * to free it again. */
+			/* Extend first to drop the refcount, such that $x .= ...; may happen in-place. */
```

Comment:
Release/free first, right?

Reply:
Yes of course :facepalm: clearly not awake anymore :p